packages/api/src/controllers/webhook.ts

```diff
@@ -237,6 +234,39 @@ app.patch(
   }
 );
 
+app.post(
+  "/:id/status",
+  authorizer({ anyAdmin: true }),
+  validatePost("webhook-status-payload"),
+  async (req, res) => {
+    const { id } = req.params;
+    const webhook = await db.webhook.get(id);
+    if (!webhook) {
+      return res.status(404).json({ errors: ["not found"] });
+    }
+
+    const doc = req.body.status;
+    const webhookResponse = req.body.response;
```

Comment:
**nit:** You can grab all the fields **with the proper typings!** like this:
```suggestion
    const { status, response} = req.body as WebhookStatusPayload;
```

You could still rename the destructured variables with sth like this `const { status: doc, response: webhookResponse}`, but I think the original names of the fields would be the clearest here anyway. `doc` kinda hides away what it is, would only use that if there's only one "payload" being handled in the function and `doc` is that.

Reply:
Hmmmmm, actually, thinking further about this, I think the clients shouldn't specify the `status` of the webhook. That because the `status` field is actually **computed** from the response:
https://github.com/livepeer/livepeer-com/blob/0f808b4b87eb72942eb098e8c3e4f13ea7699a90/packages/api/src/webhooks/cannon.ts#L456

So we would end up either missing the `lastFailure` field which is the most important for debuggability (and that's where the dashboards shows errors from), or the client would have to specify some redundant stuff in their request, with the response (which might be big 📈) going twice. So I am thinking we could do:
 - Receive only the `response` object in the request payload
 - Derive the `status` from it instead, likely reusing some of the same object in `cannon.ts` (or if you copy:
```ts
      const triggerTime = response.createdAt ?? Date.now()
      let status: DBWebhook["status"] = { lastTriggeredAt: triggerTime };
      if (response.statusCode >= 300 || !response.statusCode) {
        status = {
          ...status,
          lastFailure: {
            timestamp: triggerTime,
            statusCode: response.statusCode,
            response: response.response.body,
          },
        };
      }
      await this.db.webhook.updateStatus(response.webhookId, status);
```
)

Reply:
I guess this makes sense, but what about the cases where we don't get any response from the webhook? For example, when there is a connection reset or a failed DNS resolve, we actually don't have a status code and a response. Unless if the status code is missing then we consider it a failure and updated `LastFailure` accordingly?

Reply:
Ok I missed that it's actually handled! Sounds good!

Reply:
I think we also have an `errorMessage` field in the webhook status to account for those non-HTTP failures. We could probably receive it from the caller as well, as a separate field apart from the response (just like on the status as well afaik). WDYT?

Reply:
Cool, makes sense! I am keeping also the `errorMessage` as a separate field. `response.statusCode` is a required field tho in the payload, so I either set up the `statusCode` as 0 when the response fails or I make it optional, wdyt?

Reply:
You could have a request like this
```yaml
{
  errorMessage: foo,
  response: { status: 123, ... }
}
```

In which status(code) is a required field of response, but response is not a required field of that payload. So you can omit the whole response if you got some different error, but if you do send a response it must have a status code. WDYT?

Reply:
FTR: We decided to simplify the first version and just comment out the `webhook_response` table update. Will do a follow-up to start saving the responses soon.